src/lib/Cropper.svelte

```diff
@@ -229,34 +230,55 @@
       restrictPosition
     )
 
-    dispatch('cropcomplete', {
+    oncropcomplete?.({
       percent: croppedAreaPercentages,
       pixels: croppedAreaPixels,
     })
   }
 
   // ------ Reactive statement ------
   //when aspect changes, we reset the cropperSize
-  $: if (imgEl) {
-    cropperSize = cropSize ? cropSize : helpers.getCropSize(imgEl.width, imgEl.height, aspect)
-  }
+  $effect(() => {
+    if (imgEl) {
+      cropperSize = cropSize ? cropSize : helpers.getCropSize(imgEl.width, imgEl.height, aspect)
+    }
+  })
 
   // when zoom changes, we recompute the cropped area
-  $: zoom && emitCropData()
+  let lastZoom = $state(zoom);
+  $effect(() => {
+    if (zoom !== lastZoom) {
+      lastZoom = zoom;
+      emitCropData()
+    }
+  })
 
-  $: if(aspect) {
-    computeSizes()
-    emitCropData()
-  } 
+  let lastAspect = $state(aspect);
+  $effect(() => {
+    if (aspect !== lastAspect) {
+      lastAspect = aspect;
+      computeSizes()
+      emitCropData()
+    }
+  })
 </script>
 
 <svelte:window on:resize={computeSizes} />
 <div
   class="svelte-easy-crop-container"
   bind:this={containerEl}
-  on:mousedown|preventDefault={onMouseDown}
-  on:touchstart|nonpassive|preventDefault={onTouchStart}
```

Comment:
The nonpassive modifiers were added on purpose. 

According to the docs, we need to use actions to register the handler ourselves: ![Screenshot_2024-12-24-02-21-43-23_e4424258c8b8649f6e67d283a50a2cbc.jpg](https://github.com/user-attachments/assets/201f446c-081d-4b3e-8b4f-9b9f3b90e088)

I assume they are referring to these actions: https://svelte.dev/docs/svelte/use

Reply:
Like this?

```ts
 const containerAction: Action<HTMLDivElement> = node => {
    $effect(() => {
      node.addEventListener('touchstart', onTouchStart)
      node.addEventListener('mousedown', onMouseDown)
      node.addEventListener('wheel', onWheel)

      return () => {
        node.removeEventListener('touchstart', onTouchStart)
        node.removeEventListener('mousedown', onMouseDown)
        node.removeEventListener('wheel', onWheel)
      }
    })
  }
```

Reply:
Yes exactly 🙂